### PR TITLE
[DI] highlight suspicious tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Why doesn't this break something?
Related to https://github.com/symfony/symfony/pull/27710 I think
/cc @xabbuh @ro0NL if you have any idea why :)

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php [hunk 1]
```diff
@@ -340,6 +341,7 @@ class ConfigurationWithArrayNodeRequiringOneElement implements ConfigurationInte
 {
     public function getConfigTreeBuilder()
     {
+        throw new \Exception();
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php]: the test ensures we dont require a node to be present, if the user didnt configure anything. However the side effect is the config tree isnt even initialized ... so it doesnt matter much what happens here.

I'd keep this tree for the purpose of clarity, in the sense we don't expect an `InvalidConfigException` due `nodes` being required.

File: src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php [hunk 2]
```diff
@@ -332,6 +332,7 @@ class EnvConfigurationWithoutRootNode implements ConfigurationInterface
...
```
Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php]: this test from #27472 is actually invalidated due #27710, we should pass some config to ensure the treebuilder is called

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateEnvPlaceholdersPassTest.php]: up for a PR?

**Comment by nicolas-grekas** [comment]
Closing in favor of #31944, thanks @ro0NL

**Pull request closed without merging**